### ISSUE TITLE
Members REST API for managing users in repositories [#96516348]

### ISSUE THREAD

**Issue by pdpinch** [mitodl/lore#129]
See also https://github.com/mitodl/lore/pull/119/files#diff-2625016b50d68d922257f74801cac29cR294

Add/List/remove repository members via REST API. 
- [x] ~~Members can be added by e-mail address (typically kerberos id + mit.edu).~~ (this will be handled in the ui)
- [x] If the user doesn't exist, an error is returned
- [x] Should be limited to users with admin privileges on the repository (could be another issue if necessary)
- [x] update the API docs

**Comment by pdpinch** [comment]
@giocalitri asked some questions:

>  The admin can see any user?

Yes. There's not much point in protecting the user ids, as they're equivalent to kerberos ids and can be easily looked up. 

>    if yes, how?

For now, I think users just enter an id. If it doesn't exist in LORE, they get an error (similar to what edX does). 

We might want to add an invitation feature later. 

>    the user to add needs to login at least once before s/he can be added (to be in the local Django DB)?

I think so. 

> Should we split this in at least 2 stories:
>         api
>         views/templates 

Will do. This issue will be the API and I'll open a new one for the views/templates

**Comment by giocalitri** [comment]
this is blocked by PR #106

**Comment by giocalitri** [comment]
Do we need a new permission for this? Something like "manage_repo_users" to add only to the administrator group repo?
Otherwise what should I check to see who has permission to mange users: only the fact that the user belong to the repo admin group?
cc @Ferdi @pdpinch

**Comment by Ferdi** [comment]
It sound like an implementation detail, no ?  
Yes,  repo admins will have the right to manage users but not sure what's the best way to implement a) adding a "manage users" permission and assigning to "admin" role or b) simply checking if user belong to 'admin' role

**Comment by giocalitri** [comment]
any preference on the URL for the API?

**Comment by Ferdi** [comment]
@giocalitri https://goo.gl/qU2PIy ?

**Comment by pdpinch** [comment]
Functionally, this is done. The API docs #295 still need to be merged.
